### PR TITLE
Portfolio: details popup window

### PR DESCRIPTION
**Pull request by Umuhan**
# Hi, Code Reviewer.
## The following are the changes made:

- [x] When the user clicks the button to check project details, the popup with details about the project appears.

- [x] The popup disappears when the user clicks the close (X) button.

- [x] The use of JavaScript array to store all of the information for all projects.

- [x] The project section is created dynamically using the information stored in the JavaScript object. 

- [x] The popup window for both the mobile version and the desktop.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -216,25 +119,75 @@ <h3 class="project-title">
...
+              ></span>
+            </button>
+            <button class="btn-source">
+              See source <span><i class="fa-brands fa-github"></i></span>
+            </button>
+          </div>
+        </div>
+      </div>
+      <div id="overlay"></div>
+    </section>
```
Comment (FrederickMih) [inline index.html:175]: - [ ] Hi @Umuhan, based on the requirements of this project, this information on your pop-up should have been created dynamically from the projects array in your JS file not from your HTML file as it is the case now. All the HTML of this section should be created when the page loads. This section of the HTML codes should be removed and added dynamically via JavaScript. The reason for adding projects from the projects array is to make it very easy to add more projects in the future. Kindly adhere to the requirements.
- [ ] Again you'll make sure your design is exactly the same as in Figma. There are some differences on the position of technologies, see live, and see sources buttons. Kindly check the differences in the 
screenshots and fix them

| Figma :+1:  | Yours |
| --- | --- |
| Screenshot from 2023-05-23 13-47-50 | Screenshot from 2023-05-23 13-49-24 |

- For a good user experience, let's remove the scrolling effect. The pop-up should fill the screen
- Also, the mobile version should be exactly the same as in Figma.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @Umuhan, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- PR is professional :+1: 
- Good commit history :+1: 
- Linter checks passed :heavy_check_mark: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html (hunk 1 above)
Comment (Umuhan) [reply index.html]: done.

File: main.js [hunk 2]
```diff
@@ -15,3 +111,81 @@ const initApp = () => {
...
+  modelSection.style.display = 'block';
+  overlay.style.display = 'block';
+}
+
+function hidePopup() {
+  modelSection.style.display = 'none';
+  overlay.style.display = 'none';
+}
+
+// document.querySelector(".btn-live").addEventListener("click", showPopup);
```
Comment (OyePriscilla) [inline main.js:132]: - [optional] Please kindly remove this commented line of code, this will make your codebase clean. ℹ️

File: main.js [hunk 3]
```diff
@@ -15,3 +111,81 @@ const initApp = () => {
...
+          <div class="model-footer desktop-flex">
+            <button class="btn-live" id="live-button">
+              See live
+              <span
+                ><i
+                  class="fa-thin fa-arrow-up-left-from-circle fa-rotate-90"
+                ></i
+              ></span>
+            </button>
+            <button class="btn-source" id="source-button">
```
Comment (OyePriscilla) [inline main.js:181]: - [ ] Please as stated earlier, kindly ensure that this is generated dynamically  with JavaScript. This portion of code is still html hardcoded in js file. I think you can add other details of pop-up to your `const project` data and populate it the way you did for modal container. 👍

Comment (OyePriscilla) [reply main.js:181]: - [ ] Awesome designing the pop-up according to the design, but kindly remove the vertical scroll as suggested earlier for good user experience. **Kindly take not that the same is applicable to the mobile version as well.** 👍 

##  Your Design

Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Umuhan 👏, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Populate project cards dynamically 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.js (hunk 2 above)
Comment (Umuhan) [reply main.js]: done.

File: main.js (hunk 3 above)
Comment (Umuhan) [reply main.js]: done

File: main.js (hunk 2 above)
Comment (Umuhan) [reply main.js]: done

File: main.js [hunk 4]
```diff
@@ -15,3 +111,27 @@ const initApp = () => {
...
+  modelSection.style.display = 'block';
+  overlay.style.display = 'block';
+}
+
+function hidePopup() {
+  modelSection.style.display = 'none';
+  overlay.style.display = 'none';
+}
+
+// document.querySelector(".btn-live").addEventListener("click", showPopup);
```
Comment (Yerimah) [inline main.js]: - [ OPTIONAL] As suggested by the previous reviewer, kindly remove every commented code from your codebase in order to maintain clean code.

File: index.html [hunk 5]
```diff
@@ -216,25 +119,75 @@ <h3 class="project-title">
...
+    <!-- modal section -->
+    <section id="model-section" class="model-section">
+      <div class="model">
+        <div class="model-content">
+          <div class="model-header">
+            <button class="model-close" type="button">X</button>
+            <img src="./images/Snapshoot Portfolio (1).png" alt="Model-image" />
```
Comment (Yerimah) [inline index.html]: - [ ] As suggested by the previous reviewer, I would encourage you to remove the scrolling effect for a good user experience.

- [ ] The desktop and mobile versions should be exactly the same as the chosen design template in Figma.

> Figma
Screenshot 2023-05-26 at 11 44 07

> Your design
Screenshot 2023-05-26 at 11 46 27

**Review by Yerimah — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES 🔴 

Hi Team, 👋  

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f4b70556d45436b5038554458344c37536b552f67697068792e676966


 ## Highlights
- No linter error 👍 
- Project cards dynamically rendered 👍 
- Descriptive PR summary 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nicupop729 — COMMENTED** [review]
Hello @Umuhan  👋,

_This is an additional review as a result of the second opinion request._

Great job on completing the project! :clap:

Firstly, I observed that during the pair-programming activity, your partner only made some CSS style removals. Typically, in such milestones, collaboration should be a key factor. It is important that we do not overlook this aspect here at Microverse.

Secondly, when we receive a review that we disagree with, it is not recommended to make additional commits before requesting a Second Opinion review. I noticed that there have been three commits made after the last review that you disagreed with.

However, I have reviewed your code and with the recent changes, the project is now ready to be approved.

Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approve right away :)

Keep rocking 🚀!

**Review by uwadonat — APPROVED** [review]
## Hi @Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
approval1

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**